### ISSUE TITLE
Don't over indent in lamba

### ISSUE THREAD

**Issue by Khady** [ocaml-ppx/ocamlformat#494]
When using a pipe and a lamba function, the code of the lambda ends up very indented. I don't understand why exactly. It doesn't happen without the pipe. Would be nice to avoid this extra indentation.

```ocaml
(* original with pipe *)
let () =
  values
  |> List.iter begin fun x ->
    let x = " " ^ x in
    print_endline x
  end

(* ocamlformat with pipe, bad indentation *)
let () =
  values
  |> List.iter (fun x ->
         let x = " " ^ x in
         print_endline x )

(* original no pipe *)
let () =
  List.iter begin fun x ->
    let x = " " ^ x in
    print_endline x
  end values

(* ocamlformat no pipe, good indentation *)
let () =
  List.iter
    (fun x ->
      let x = " " ^ x in
      print_endline x )
    values
```

**Comment by jberdine** [comment]
The indentation of the `fun` body should be indented the same offset, 4, from the beginning of the application (here, the beginning of `List.iter`) irrespective of using `|>`, and irrespective of whether there is a break before `(fun`.

**Comment by brendanlong** [comment]
I'm not sure if this is complicated to implement, but the "before" code above is matching 100% of the existing OCaml code I've read, and 4 or 6 space indentation are both changes that look really inconsistent. I'm trying to switch some code over to ocamlformat, but it's a really big and unwelcome change that suddenly some things are indented by 2 spaces, some indented by 4 spaces, and some are indented by 6 spaces.

Is there any chance of a mode where every time we indent a new line, we just indent by 2 spaces more than the previous indentation level 100% of the time?

**Comment by jberdine** [comment]
Closing this as there isn't anything planned that is similar to a increase indentation ignoring expression structure.

**Comment by gpetiot** [comment]
Fixed with #841

**Comment by Khady** [comment]
Thank you!
